status.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/hansmi/paperhooks/pkg/client"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type statusClient interface {
+	GetStatus(ctx context.Context) (*client.SystemStatus, *client.Response, error)
+}
+
+type statusCollector struct {
+	cl statusClient
+
+	storageTotalDesc                *prometheus.Desc
+	storageAvailableDesc            *prometheus.Desc
+	databaseStatusDesc              *prometheus.Desc
+	databaseUnappliedMigrationsDesc *prometheus.Desc
+	redisStatusDesc                 *prometheus.Desc
+	celeryStatusDesc                *prometheus.Desc
+	indexStatusDesc                 *prometheus.Desc
+	indexLastModifiedDesc           *prometheus.Desc
+	classifierStatusDesc            *prometheus.Desc
+	classifierLastTrainedDesc       *prometheus.Desc
+}
+
+// Only doing this to be able to unit test the collector
+var timeSince = time.Since
+
+func newStatusCollector(cl statusClient) *statusCollector {
+	return &statusCollector{
+		cl: cl,
+
+		storageTotalDesc:                prometheus.NewDesc("paperless_status_storage_total", "Total storage of Paperless in bytes.", nil, nil),
+		storageAvailableDesc:            prometheus.NewDesc("paperless_status_storage_available", "Available storage of Paperless in bytes.", nil, nil),
+		databaseStatusDesc:              prometheus.NewDesc("paperless_status_database_status", "Status of the database. 1 is OK, 0 is not OK.", nil, nil),
+		databaseUnappliedMigrationsDesc: prometheus.NewDesc("paperless_status_database_unapplied_migrations", "Number of unapplied database migrations.", nil, nil),
+		redisStatusDesc:                 prometheus.NewDesc("paperless_status_redis_status", "Status of redis. 1 is OK, 0 is not OK.", nil, nil),
+		celeryStatusDesc:                prometheus.NewDesc("paperless_status_celery_status", "Status of celery. 1 is OK, 0 is not OK.", nil, nil),
+		indexStatusDesc:                 prometheus.NewDesc("paperless_status_index_status", "Status of the index. 1 is OK, 0 is not OK.", nil, nil),
+		indexLastModifiedDesc:           prometheus.NewDesc("paperless_status_index_last_modified", "Seconds since the last time the index has been modified.", nil, nil),
+		classifierStatusDesc:            prometheus.NewDesc("paperless_status_classifier_status", "Status of the classifier. 1 is OK, 0 is not OK.", nil, nil),
+		classifierLastTrainedDesc:       prometheus.NewDesc("paperless_status_classifier_last_trained", "Seconds since the last time the classifier has been trained.", nil, nil),
```

Comment:
Prometheus metrics [are strongly recommended to use absolute timestamps](https://prometheus.io/docs/practices/instrumentation/#timestamps-not-time-since). This metric would be `paperless_status_classifier_last_trained_timestamp_seconds`. Example:

https://github.com/hansmi/prometheus-paperless-exporter/blob/a4a892127ca521e79d3a9edafba73bbf3cbb7c45/correspondent.go#L33-L35

Reply:
Fixed in https://github.com/hansmi/prometheus-paperless-exporter/pull/58/commits/b0972dfd4bc1751f982e9ea5dfbf93dce0d90c57